src/main/java/org/junit/Assert.java

```diff
@@ -955,4 +955,48 @@ public static void assertEquals(Object[] expecteds, Object[] actuals) {
             Matcher<? super T> matcher) {
         MatcherAssert.assertThat(reason, actual, matcher);
     }
+
+    /**
+     * This interface facilitates the use of expectThrows from Java 8. It allows method references
+     * to void methods (that declare checked exceptions) to be passed directly into expectThrows
+     * without wrapping. It is not meant to be implemented directly.
+     *
+     * @since 4.13
+     */
+    public interface ThrowingRunnable {
+        void run() throws Throwable;
+    }
+
+    /**
+     * Asserts that {@code runnable} throws an exception of type {@code throwableClass} when
+     * executed. If it does, the exception object is returned. If it does not throw an exception, an
+     * {@link AssertionError} is thrown. If it throws the wrong type of exception, an {@code
+     * AssertionError} is thrown describing the mismatch; the exception that was actually thrown can
+     * be obtained by calling {@link AssertionError#getCause}.
+     *
+     * @param throwableClass the expected type of the exception
+     * @param runnable       A function that is expected to throw an exception when executed
+     * @return The exception thrown by {@code runnable}
+     * @since 4.13
+     */
+    public static <T extends Throwable> T expectThrows(Class<T> throwableClass, ThrowingRunnable runnable) {
+        try {
+            runnable.run();
+        } catch (Throwable t) {
+            if (throwableClass.isInstance(t)) {
```

Comment:
@rschmitt sorry if this has been frustrating.

Firstly, I apologize for not making it clear in my longer comment above that I think we should include an API like this in JUnit soon (and not wait until JUnit Lamba). While I understand the reasons why some people want to wait, I don't find the arguments presented so far convincing. I don't want to dismiss people's concerns out of hand, so I'm still leaving the door open for someone to convince me that we should wait.

Secondly, I ask that you please have patience in our process. Designing an API takes time. The size of the API is often unrelated to the amount of time it might take to ensure it's what we want to build, and make sure it's polished and well documented.

If we were developing an API that was going to be used by only a hundred people at one company, I'd be fine with what you proposed here, but by some estimates JUnit is used by millions of developers (based [this](https://www.google.com/search?q=how+many+java+developers+worldwide) and [this](http://zeroturnaround.com/rebellabs/java-tools-and-technologies-landscape-for-2014)). Taking a bit of time to ensure that this is the API that we want our developers to use is time worth spent.

Your original pull request was opened *only two days ago*, so I think it may be a bit of a stretch to say we are dragging our feet :-) 

Finally, I think you should take the comments from people as evidence of how enthusiastic people are to have an API like this as part of JUnit. We're simply trying to come to a consensus on what that API should look like, and when and how to release it.